docker/BUILD.bazel

```diff
@@ -12,6 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+genrule(
+    name = "orchestration_image_dev_tar",
+    outs = ["orchestration-image-dev.tar"],
+    cmd = "$(location :orchestration_image_builder) -o $@ -d",
```

Comment:
The difference between generating a `dev` image and a `prod` image is by appending `-d` in the command line which is then handled in two different bash scripts. I find this error prone, if we remove `-d` by mistake or introduce a bug in the scripts, it will create major problems down the line, for example not testing against latest changes allowing new PRs to be merged in.

I'd like to have a less error prone alternative here. My suggestion is  to only build `dev` image with `bazel`. Building the image with `bazel` was created only from the e2etests perspective as it makes running the e2e tests very simple using only one command. Now, the way the image is built with `bazel` is still a hacky way, using a wrapper bash script. Hence I prefer to avoid extending/accommodating  this solution to build the prod image and use the bash script directly.

Reply:
I agree with `prod` image `bazel` rule wouldn't be needed for performing e2e testing of cuttlefish host packages, but I think probably `prod` image `bazel` rule is needed for performing e2e testing of `cvdr`. Because versioning won't be same between the docker image and cuttlefish host packages. In case of modifying Dockerfile, performing `cvdr` e2e testing may not catch the error specified at `prod` image, and it can be mattered when releasing new version of `prod` image.

If you're still worried about error prone, what about explicitly giving parameter that this is for `dev` or `prod` in the script?

Reply:
@ser-io Gentle ping.

Reply:
Ideally we should be using bazel rule https://github.com/bazelbuild/rules_docker/blob/master/docs/container.md#container_image to create Docker Image artifacts with `bazel`. I created the bash wrapper script hack to enable the e2etests using bazel, let's not extend this solution to build prod images as it's not ideal.

I understand the case of `cvdr`, however it's not clear to me how we will be using the rule in the e2e tests for `cvdr` as `cvdr` is in a different repository, we don't even use bazel in the `cvdr` repo yet. We can revisit this idea when implementing `cvdr` e2e tests, but let's wait until then and keep it simple for now.

Reply:
Understood. I think not using bazel wrapper script for `prod` image is fine for now, but I'm still afraid of docker images between `prod` and `dev` image are diverged with unexpected behavior. So please let me know if there's a future plan for modifying bazel wrapper script.

OTOH, if we use `container_image` bazel rule someday, I think `prod` image should be built with that as well so far due to the above reason.

I'll be changing PR without using bazel wrapper for `prod` image.

Reply:
Done.

Reply:
Yes, if there's a use case to build the `prod` image with `bazel` we should aim to build it with `container_image`.